### PR TITLE
npm changed to nvm oops

### PR DESCRIPTION
**Pull request by timjonesplaysports**

### CODE REVIEW

**Comment by ashsmith** [comment]
Just to weigh in... A node runtime won't be needed by every team. Is it worth considering splitting these into multiple images? We could take advantage of [docker multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) here, or maintain multiple docker files for more specific runtimes. That way we could avoid this becoming a monolithic Docker image that overtime takes longer to be pulled by CI due to its increasing size.

We could have some resulting images named like so:

```
playsportsgroup/buildtools-node:latest
playsportsgroup/buildtools-php:latest
playsportsgroup/buildtools-ruby:latest
```

Each one would build from a common base (playsportsgroup/buildtools:latest for example), then only adding in specific tools needed to each image.

**Comment by timjonesplaysports** [comment]
I completely understand this and did question this already. 

But what my difficulty is: Time.  I'd need to spend time doing this and I have other priorities that need attention.

**Comment by ashsmith** [comment]
Cool, I'm happy for this to be merged for the time being. I'll pick up building separate images when I've got some downtime.

File: Dockerfile [hunk 1]
```diff
@@ -2,10 +2,15 @@ FROM alpine
 ARG CLOUD_SDK_VERSION=254.0.0
 ENV CLOUD_SDK_VERSION=$CLOUD_SDK_VERSION
 
-RUN apk add --no-cache curl python python3 py-crcmod py-pip python-dev libffi-dev bash libc6-compat openssh-client openssl-dev git gnupg rsync coreutils gcc libc-dev make nodejs npm
-RUN npm install 8
-RUN npm install 10
-RUN npm install 12
+RUN apk add --no-cache curl python python3 py-crcmod py-pip python-dev libffi-dev bash libc6-compat openssh-client openssl-dev git gnupg rsync coreutils gcc libc-dev make npm ca-certificates ncurses g++ libgcc linux-headers grep util-linux binutils findutils
```
Comment (ashsmith) [inline Dockerfile]: Do we need all of these extra tools in the image, or do we need them just for installing NVM?

To optimise this we can install them temporarily by saving them to a virtual package, then removing the virtual package. 

```
RUN apk add --no-cache --virtual nvm-build-deps \
  ca-certificates ncurses g++ libgcc linux-headers grep util-linux binutils findutils && \
  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash && \
  apk remove --virtual nvm-build-deps
```

Additionally, by keeping everything in one docker instruction we keep the resulting image size down. As each instruction is a layer in the docker image, which means when something is added in one layer, the size increases, even if you remove it in a later layer.

We can further optimise this by reducing the total number of layers too, as well as removing anything we don't need once the image is built.

**Pull request merged**